src/wall_segment_sim.cpp

```diff
@@ -83,7 +83,7 @@ bool WallSegmentSim::Trace(double x, double y, double theta, double length, doub
   if (hits.size() > 0) {
     for (auto hit : hits) {
       double hit_chance = hit.second->max_return;
-
+      intensity = hit.second->intensity;
```

Comment:
should be placed after line 97

Reply:
doesn't matter. the intensity get is just set here. It is published in `ray_cast` if there was a hit

Reply:
wait let me think if there is an edge case

Reply:
It does matter, this can not return that specific hit if it doesn't meet the hit chance

Reply:
also you don't want high intensity beams that are at max distance

Reply:
true. The only reason I didn't see it is because we update intensity only for "obstacles".